aws-logs-destination/src/test/java/software/amazon/logs/destination/UpdateHandlerTest.java

```diff
@@ -182,4 +182,45 @@ public void handleRequest_Should_ReturnSuccess_When_DestinationIsFound() {
         assertThat(response.getErrorCode()).isNull();
     }
 
+    // tests for optional parameter destination policy not provided tests
+    @Test
+    public void handleRequest_Should_ReturnSuccess_When_DestinationIsFound_and_DestinationPolicyNotProvided() {
```

Comment:
Can we add a test for DestinationIsFound, DestinationPolicyNotProvided, and the initial destination had a destination policy?

Reply:
Hmm, "initial destination had a destination policy" implying the resource is created with the destination policy but the update doesn't have a destination policy? If that is what you mean, that case is in the integration testing. AFAIU, unit tests only test one function, so none of the tests in these files call more than one function.

Reply:
Never mind, I see what you mean, added test case

Reply:
was this supposed to be DestinationNotFound? if the DestinationIsFound doesnt that mean the resource already exist and should fail?

Reply:
This is the handler for update. I think you are confusing this with create. Create fails if the destination already exists. Update should fail in the case that destination does not exist, and should not fail when the destination does exist(given  of course there aren't any other unrelated errors).